Critical: Linux/Windows CLI installer fails due to restrictive file size validation

## 🚨 Critical Bug: CLI Installer Broken for Linux and Windows Users

### Problem
Users on Linux and Windows cannot install the CLI via the install script. The installer fails with:
```
[ERROR] Downloaded file size (109608998 bytes) outside expected range (10MB-100MB)
install: cannot stat '': No such file or directory
```

### Root Causes

#### 1. File Size Validation Too Restrictive
- **Location**: `packages/cli/scripts/install.sh` line 191
- **Current check**: Files must be between 10MB-100MB
- **Actual sizes in v0.6.6**:
  - ✅ macOS ARM64: 66MB (passes)
  - ✅ macOS x64: 72MB (passes)
  - ❌ **Linux x64: 109.6MB (FAILS)**
  - ❌ **Windows x64: 124.8MB (FAILS)**

#### 2. Poor Error Handling
- **Location**: `packages/cli/scripts/install.sh` line 440
- When `download_binary` fails in subshell via `$()`, the error doesn't propagate
- Script continues with empty `$temp_binary` variable
- Causes secondary error: `install: cannot stat ''`

### Impact
- **Affected versions**: All releases since at least v0.3.14 (checked back to March 2024)
- **Affected platforms**: All Linux and Windows users
- **Severity**: CRITICAL - Complete installation failure for 2/3 of supported platforms

### Historical Data
Linux binary sizes have been consistently over 100MB:
- v0.3.14: 103MB
- v0.4.x: 103MB  
- v0.5.x: 104MB
- v0.6.x: 104MB

This means the Linux installer has NEVER worked with the current size validation!

### Fixes Required

#### Fix 1: Update Size Limit
```bash
# Line 191: Change from 100MB to 150MB
- if [ "$file_size" -lt 10485760 ] || [ "$file_size" -gt 104857600 ]; then
+ if [ "$file_size" -lt 10485760 ] || [ "$file_size" -gt 157286400 ]; then

# Line 193: Update error message
- print_error "Downloaded file size ($file_size bytes) outside expected range (10MB-100MB)"
+ print_error "Downloaded file size ($file_size bytes) outside expected range (10MB-150MB)"
```

#### Fix 2: Improve Error Handling
```bash
# After line 440, add:
local temp_binary=$(download_binary "$version" "$platform")
+ [ -z "$temp_binary" ] && exit 1
```

#### Fix 3: Consider PowerShell Script
The PowerShell installer (`install.ps1`) doesn't have size validation. Consider adding it for consistency, but with the corrected 150MB limit.

### Test Case
```bash
# Reproduce the issue
curl -fsSL https://raw.githubusercontent.com/cogflows/promptcode-vscode/main/packages/cli/scripts/install.sh | bash

# After fix, should successfully install on Linux
```

### User Report
Originally reported by user on Linux:
- Platform: `Linux fw 6.14.0-27-generic #27-Ubuntu x86_64 GNU/Linux`
- Full error output provided in description

cc: @aviv-s

--- a/.claude/settings.local.json
+++ b/.claude/settings.local.json
@@ -206,7 +206,8 @@
       "Bash(kill:*)",
       "Bash(tty)",
       "Bash(timeout:*)",
-      "Bash(pkill:*)"
+      "Bash(pkill:*)",
+      "Bash(gh issue:*)"
     ],
     "deny": [],
     "defaultMode": "acceptEdits",

--- a/.github/workflows/test-installer.yml
+++ b/.github/workflows/test-installer.yml
@@ -1,0 +1,143 @@
+name: Test Installer Scripts
+
+on:
+  pull_request:
+    paths:
+      - 'packages/cli/scripts/install.sh'
+      - 'packages/cli/scripts/install.ps1'
+      - '.github/workflows/test-installer.yml'
+  push:
+    branches:
+      - main
+      - develop
+    paths:
+      - 'packages/cli/scripts/install.sh'
+      - 'packages/cli/scripts/install.ps1'
+      - '.github/workflows/test-installer.yml'
+  workflow_dispatch:  # Allow manual testing
+  release:
+    types: [published]
+
+jobs:
+  test-bash-installer:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Test install.sh syntax
+        run: |
+          bash -n packages/cli/scripts/install.sh
+          echo "✓ Bash syntax check passed"
+      
+      - name: Test installer with latest release
+        run: |
+          # Test the installer script
+          export PROMPTCODE_INSTALL_DIR=$(mktemp -d)
+          echo "Installing to: $PROMPTCODE_INSTALL_DIR"
+          
+          # Run installer and capture output
+          if ! bash packages/cli/scripts/install.sh 2>&1 | tee install.log; then
+            echo "ERROR: Installation failed"
+            cat install.log
+            exit 1
+          fi
+          
+          # Check if binary was installed
+          if [ ! -f "$PROMPTCODE_INSTALL_DIR/promptcode" ]; then
+            echo "ERROR: Binary not found at $PROMPTCODE_INSTALL_DIR/promptcode"
+            cat install.log
+            exit 1
+          fi
+          
+          # Test the installed binary - CRITICAL: Must not hang
+          echo "Testing binary with --version (should not hang)..."
+          # Use perl for cross-platform timeout (works on both Linux and macOS)
+          if ! perl -e 'alarm shift; exec @ARGV' 10 "$PROMPTCODE_INSTALL_DIR/promptcode" --version; then
+            echo "ERROR: promptcode --version either failed or hung (timeout after 10s)"
+            exit 1
+          fi
+          
+          # Test help command as well (just make sure it doesn't hang)
+          echo "Testing binary with --help..."
+          perl -e 'alarm shift; exec @ARGV' 10 "$PROMPTCODE_INSTALL_DIR/promptcode" --help > /dev/null 2>&1
+          echo "✓ --help command completed without hanging"
+          
+          echo "✓ Installation and sanity tests successful on ${{ matrix.os }}"
+      
+
+  test-powershell-installer:
+    runs-on: windows-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Test install.ps1 syntax
+        shell: pwsh
+        run: |
+          # PowerShell syntax check
+          $null = [System.Management.Automation.Language.Parser]::ParseFile(
+            "packages/cli/scripts/install.ps1",
+            [ref]$null,
+            [ref]$null
+          )
+          Write-Host "✓ PowerShell syntax check passed"
+      
+      - name: Test installer with latest release
+        shell: pwsh
+        run: |
+          # Test the installer script
+          $env:PROMPTCODE_INSTALL_DIR = New-TemporaryFile | % { Remove-Item $_; New-Item -ItemType Directory -Path $_ }
+          Write-Host "Installing to: $env:PROMPTCODE_INSTALL_DIR"
+          
+          # Run installer (with -Insecure for CI testing, since checksum might fail)
+          try {
+            & packages/cli/scripts/install.ps1 -Insecure
+          } catch {
+            Write-Error "Installation failed: $_"
+            exit 1
+          }
+          
+          # Check if binary was installed
+          $binaryPath = Join-Path $env:PROMPTCODE_INSTALL_DIR "promptcode.exe"
+          if (!(Test-Path $binaryPath)) {
+            Write-Error "Binary not found at $binaryPath"
+            exit 1
+          }
+          
+          # Test the installed binary - CRITICAL: Must not hang
+          Write-Host "Testing binary with --version (should not hang)..."
+          $job = Start-Job -ScriptBlock { param($path) & $path --version } -ArgumentList $binaryPath
+          if (!(Wait-Job $job -Timeout 10)) {
+            Stop-Job $job
+            Remove-Job $job
+            Write-Error "promptcode --version hung (timeout after 10s)"
+            exit 1
+          }
+          $output = Receive-Job $job
+          Remove-Job $job
+          Write-Host "Version output: $output"
+          
+          # Test help command as well
+          Write-Host "Testing binary with --help..."
+          $job = Start-Job -ScriptBlock { param($path) & $path --help } -ArgumentList $binaryPath
+          if (!(Wait-Job $job -Timeout 10)) {
+            Stop-Job $job
+            Remove-Job $job
+            Write-Error "promptcode --help hung (timeout after 10s)"
+            exit 1
+          }
+          $helpOutput = Receive-Job $job
+          Remove-Job $job
+          # Convert array to string and check if it contains promptcode
+          $helpText = $helpOutput -join " "
+          if ($helpText -notmatch "promptcode") {
+            Write-Error "Help output doesn't contain expected text: $helpText"
+            exit 1
+          }
+          
+          Write-Host "✓ Installation and sanity tests successful on Windows"
+

--- a/packages/cli/scripts/install.ps1
+++ b/packages/cli/scripts/install.ps1
@@ -69,7 +69,7 @@ function Get-LatestVersion {
 
 # Download binary
 function Download-Binary($version, $arch) {
-    $binaryName = "${CLI_NAME}-windows-${arch}.exe"
+    $binaryName = "${CLI_NAME}-win-${arch}.exe"
     $downloadUrl = "https://github.com/$REPO/releases/download/$version/$binaryName"
     $checksumUrl = "https://github.com/$REPO/releases/download/$version/$binaryName.sha256"
     $tempFile = [System.IO.Path]::GetTempFileName() + ".exe"
@@ -270,7 +270,7 @@ function Install-PromptCode {
             
             # For dev versions, inform about --force option
             if ($currentVersion -like "*-dev.*") {
-                Write-Info "Development version detected. To force update to $version:"
+                Write-Info "Development version detected. To force update to ${version}:"
                 Write-Host ""
                 Write-Host "  $CLI_NAME update --force" -ForegroundColor Cyan
                 Write-Host ""

--- a/packages/cli/scripts/install.sh
+++ b/packages/cli/scripts/install.sh
@@ -186,12 +186,6 @@ download_binary() {
     print_error "Downloaded file appears to be HTML/text, not a binary"
   fi
   
-  # Check file size (between 10MB and 100MB)
-  local file_size=$(stat -f%z "$temp_file" 2>/dev/null || stat -c%s "$temp_file" 2>/dev/null || echo 0)
-  if [ "$file_size" -lt 10485760 ] || [ "$file_size" -gt 104857600 ]; then
-    rm -f "$temp_file"
-    print_error "Downloaded file size ($file_size bytes) outside expected range (10MB-100MB)"
-  fi
   
   # Download and verify checksum (MANDATORY for security)
   local checksum_url="${download_url}.sha256"
@@ -438,6 +432,12 @@ main() {
 
   # Download binary
   local temp_binary=$(download_binary "$version" "$platform")
+  
+  # Check if download was successful
+  if [ -z "$temp_binary" ] || [ ! -f "$temp_binary" ]; then
+    print_error "Failed to download binary"
+    exit 1
+  fi
 
   # Install binary
   install_binary "$temp_binary"

